node-repository/src/test/java/com/yahoo/vespa/hosted/provision/restapi/v2/filter/FilterTester.java

```diff
@@ -0,0 +1,168 @@
+import com.yahoo.jdisc.http.servlet.ServletRequest;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x509.BasicConstraints;
+import org.bouncycastle.asn1.x509.Extension;
+import org.bouncycastle.cert.X509v3CertificateBuilder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URI;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Date;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author mpolden
+ */
+public class FilterTester {
+
+    private final SecurityRequestFilter filter;
+
+    public FilterTester(SecurityRequestFilter filter) {
+        this.filter = filter;
+    }
+
+    public void assertSuccess(Request request) {
+        assertFalse("No response written by filter", getResponse(request).isPresent());
+    }
+
+    public void assertRequest(Request request, int status, String body) {
+        Optional<Response> response = getResponse(request);
+        assertTrue("Expected response from filter", response.isPresent());
+        assertEquals("Response body", body, response.get().body);
+        assertEquals("Status code", status, response.get().status);
+    }
+
+    private Optional<Response> getResponse(Request request) {
+        RequestHandlerTestDriver.MockResponseHandler handler = new RequestHandlerTestDriver.MockResponseHandler();
+        filter.filter(toDiscFilterRequest(request), handler);
+        return Optional.ofNullable(handler.getResponse())
+                       .map(response -> new Response(response.getStatus(), handler.readAll()));
+    }
+
+    private static DiscFilterRequest toDiscFilterRequest(Request request) {
+        DiscFilterRequest r = mock(DiscFilterRequest.class);
+        when(r.getMethod()).thenReturn(request.method().name());
+        when(r.getUri()).thenReturn(URI.create("http://localhost").resolve(request.path()));
+        when(r.getRemoteAddr()).thenReturn(request.remoteAddr());
+        if (request.commonName().isPresent()) {
+            X509Certificate cert = certificateFor(request.commonName().get(), keyPair());
+            when(r.getAttribute(ServletRequest.JDISC_REQUEST_X509CERT))
```

Comment:
Remember to use the new getter after merging #5306

Reply:
Yes, will update it in a separate PR, thanks!